src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -367,10 +367,13 @@ private static object CreateInstance(Type type)
                 return Array.CreateInstance(typeInfo.GetElementType(), 0);
             }
 
-            bool hasDefaultConstructor = typeInfo.DeclaredConstructors.Any(ctor => ctor.IsPublic && ctor.GetParameters().Length == 0);
-            if (!hasDefaultConstructor)
+            if (!typeInfo.IsValueType)
```

Comment:
Do we need to disallow internal value types that a caller wouldn't normally have access to?  We probably don't need to be any more restrictive than we are for classes, but I wasn't sure what was allowed for classes.

Reply:
Looks like this code already allows creation of an internal type in another assembly if that type had a public constructor, so it looks like this doesn't make that any worse.